### ISSUE TITLE
Promote AbstractUidType private methods to protected (Doctrine bridge)

### ISSUE THREAD

**Issue by Arkemlar** [symfony/symfony#58263]
### Description

I use uid component and have ULID class that extending Symfony's ULID one.
Now I want to have mine own UlidType implementation that extends AbstractUidType. I want to reuse it's logic as much as possible without copying all of the stuff, but it seems that I have to copy because AbstractUidType has private methods and I can't use them inside child class.
Why are they private but not protected? Lets make them protected plz.

### Example

[AbstractUidType link in repo](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Bridge/Doctrine/Types/AbstractUidType.php)

Private methods that needs to be protected:
```php
abstract class AbstractUidType extends Type
{
    // ...
    private function hasNativeGuidType(AbstractPlatform $platform): bool {}
    private function throwInvalidType(mixed $value): never {}
    private function throwValueNotConvertible(mixed $value, \Throwable $previous): never {}
}
```

**Comment by xabbuh** [comment]
We are always reluctant to turn private methods into protected ones as they make dealing with BC quite hard. Can you be more precise on your actual use case (for example, how your own type would finally look like)?

**Comment by Arkemlar** [comment]
Well, after some tests I decided to not use `AbstractUidType` as base at all. The only things that I had to copy-paste was `throwInvalidType` and `throwValueNotConvertible` methods.
My original idea was to explicitly specify binary database type for ulid (bytea in Postgres). But it was bad idea because in that case database returned resource as value and it involves too much manipulation to process stream to get string to get ulid... Interesting enough how it supposed to work because `AbstractUidType::convertToDatabaseValue` expects string but not resource.
Basically, this might be closed.
